The first quantum networks may not be "routing and forwarding" networks.

ngym opened the pull request:
Bell pairs are undirected network resources. This fact causes many differences between quantum network architectures and classical network architectures. It should be worth clarifying.

ngym committed:
add texts to explain routing and swapping network

ngym committed:
corresponding txt file. For routing and swapping

ngym committed:
fix some texts

ngym committed:
improve English

ngym committed:
improve English

ngym committed:
fix typo

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -656,6 +676,11 @@
          other schemes the nodes retain greater control over the entangled pair
          generation.</t>
 
```
I'm not sure what is actually meant by the paragraph below. What is residual entanglement? Do you mean that whilst photons are transmitted in a directional manner, the resulting entangled pair is not directed?

ngym replied:
You are correct. I mean that while photons are transmitted in a directional manner, the resulting entangled pair is not directed.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +820,47 @@
            information via separate classical channels which the repeaters will
            have to correlate with the qubits stored in their memory.</t>
 
```
This title is confusing. Maybe something simpler? E.g. "First generations of quantum networks will not be store-and-forward" perhaps?

ngym replied:
"First generations of quantum networks will not be store-and-forward" is fine.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +820,47 @@
+           <t> The first quantum networks may not be "routing and forwarding"
+           networks.
+           <vspace blankLines="1" />
+
```
This paragraph needs a bit more detail on what you mean by this difference.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +820,47 @@
+           This phenomenological distinction should result in architectural
+           differences between quantum networks and classical networks.
+           <vspace blankLines="1" />
+
```
If routing is the same, why not leave it out? Only contrast the data plane activity. Otherwise, I was reading expecting some explanation of how routing will be different, but this is not the point of this paragraph.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +820,47 @@
+           Additionally, "routing and swapping" quantum networks are not
+           "store and forward" networks.
+           <vspace blankLines="1" />
+
```
The first sentence is too detailed in technical details. Since entanglement swapping is already explained in this document, why not refer to it?

ngym replied:
I agree.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +820,47 @@
+           state among four qubits over three nodes and then measures
+           two qubits in the middle node to leave a Bell pair between
+           two distant nodes. In this sense, "routing and swapping" networks
+           also belong to "routing and measuring" networks.
```
Is "routing and measuring" that important to mention? It feels like a side-note and I think it's best to focus on the core ideas in this document

ngym replied:
"Routing and measuring" includes more generations. Hence,  routing and measuring can be omitted to focus on the first generations. Adding a section to explain the three generations will work well.

ngym replied:
Finally, I removed discussion related to measuring.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +820,47 @@
+           have been proposed. "Routing and measuring" is a broader category,
+           and more types of quantum networks will be categorized into it.
+           <vspace blankLines="1" />
+
```
Either this paragraph needs to be much clearer, or, if we add a section on the three generations, this can go there.

ngym replied:
Yep, I'll think about adding the section.

Wojtek242 reviewed (changes requested):
First, thank you for your contribution!

Overall, I think your contribution makes a very important point that will be very useful for people coming from a classical background. My comments are mostly about clarifying what you meant. 

I am also not sure about this "routing and forwarding" and "routing and swapping" terminology. Is it common? I know of "store-and-forward" and perhaps we can say "store-and-swap" which highlights the need for memory (some proposals do not need a memory, but most practical implementations so far rely on memory as far as I know). Also "store-and-swap" suggests we can keep things in memory for a long time which is not the case. I think this would be a good point to discuss. I would definitely suggest leaving routing completely out of this section as it is not actually discussed at all.

Wojtek242 commented:
I think I messed up the comments - they're meant to refer to the text below, not above - hope it's clear from the context though.

ngym committed:
reflect wojtek's comments on Apr 3. https://github.com/Wojtek242/draft-irtf-qirg-principles/pull/5

ngym committed:
Merge branch 'wojtek's master'

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -160,6 +160,26 @@
+         <country>Japan</country>
+       </postal>
+
+       <email>marcello.caleffi@unina.it</email>
```
I don't think this is your e-mail address :)

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -656,9 +676,14 @@
+         entanglements are undirected resources. Therefore we can choice
+         any type of link generation and any direction of photon transmission,
+         depending on policy, or for optimization.</t>
+
```
Hmm, this still isn't super clear to me. How about:
```suggestion
         <t> Note that whilst photons are emitted in a particular direction the 
         resulting entangled pair of qubits does not have a direction associated 
         with it. Physically, there is no upstream or downstream end of the 
         pair.</t>   
```
The second sentence in your original paragraph, whilst correct, I could not see how it supported the first sentence. If this connection was important, could you explain it?

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +920,37 @@
+
+           As described in <xref target="elg"></xref>,
+           Quantum links provide Bell pairs that are undirected network
+           resources, in contrast to directed links of classical networks.
```
```suggestion
           resources, in contrast to directed packets of classical networks.
```

Wojtek242 replied:
links don't have to be directional in classical networks either - it is packets/frames/datagrams that have a source and destination

ngym replied:
```suggestion
           resources, in contrast to directed frames of classical networks.
```
I think this should be frames because this sentence talk about links.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +920,37 @@
+           to the other end hop by hop.
+           <vspace blankLines="1" />
+
+           Classical networks store data coming from an interface, then forward
+           the data to another appropreate interface.
```
```suggestion
           Classical networks receive data on one interface, store it on local buffers, 
           then forward the data to another appropriate interface.
```

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +920,37 @@
+
+           Classical networks store data coming from an interface, then forward
+           the data to another appropreate interface.
+           Quantum networks store a couple of half Bell pairs, then
```
```suggestion
           Quantum networks store Bell pairs and then
```
I'd say by now the reader should know we mean a single node stores only one qubit of each pair (it's in the entanglement swapping section)

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +920,37 @@
+           Classical networks store data coming from an interface, then forward
+           the data to another appropreate interface.
+           Quantum networks store a couple of half Bell pairs, then
+           execute entanglement swapping instead of forwarding in a data plane.
```
```suggestion
           execute entanglement swapping instead of forwarding in the data plane.
```

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +920,37 @@
+           execute entanglement swapping instead of forwarding in a data plane.
+           Such quantum networks are "store and swap" networks.
+           In "store and swap" networks, we do not need to care about
+           the order of generating link Bell pairs since Bell pairs are
+           undirected resources. This distinction makes control algorithms and
```
```suggestion
           the order in which the Bell pairs were generated since they are
           undirected. This distinction makes control algorithms and
```

ngym replied:
```suggestion
           the order in which the Bell pairs are generated since they are
           undirected. This distinction makes control algorithms and
```

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -760,6 +785,106 @@
+     <section title="Generations of Quantum Networks">
+
+       <t>Quantum networks are categorized into three generations
+       <xref target="mural14"></xref>. Those generations are defined by
```
I don't think this citation is correct.

ngym replied:
Ooops, it should be sci. repo. in 2016.

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +920,37 @@
+           optimization of quantum networks different from classical ones.
+           <vspace blankLines="1" />
+
+           Meanwhile, third generations of quantum networks of direct transmission
+           are "store and forward" networks, since such quantum networks can carry
+           arbitrary states hop by hop. Hence those later generations of
+           quantum networks would have more in common with classical networks
+           than the first quantum networks in networking algorithms.</t>
+           
```
```suggestion
           It is worth noting, that third generation <refer to generations section> 
           quantum repeaters will be able to transmit qubits directly
           in a "store and forward" manner since they will be able to carry
           arbitrary states hop by hop. Thus, the networking algorithms of the
           later generations of quantum networks may end up having more in 
           common with classical networks.</t>
           
```

Wojtek242 reviewed (commented):
Thanks for implementing the changes. They look good to me!

I would like to spend some more time on the generation section though. I need to think more about what this section should contribute to the document. There is nothing wrong with what you wrote, I'd just rather not include it until I'm clear on what its contents should be and then I'll come back to your contribution.

However, the text about directionality, once you have a look at my proposed changes, can already be included. Since the generation section is currently in the same PR, if you would like your contributions merged sooner, please submit the section about generations as a separate PR, and I will merge the rest.

ngym committed:
Update draft-irtf-qirg-principles-03.xml

Co-Authored-By: Wojciech Kozlowski <Wojtek242@users.noreply.github.com>

ngym committed:
Update draft-irtf-qirg-principles-03.xml

Co-Authored-By: Wojciech Kozlowski <Wojtek242@users.noreply.github.com>

ngym committed:
Update draft-irtf-qirg-principles-03.xml

Co-Authored-By: Wojciech Kozlowski <Wojtek242@users.noreply.github.com>

ngym committed:
Update draft-irtf-qirg-principles-03.xml

Co-Authored-By: Wojciech Kozlowski <Wojtek242@users.noreply.github.com>

ngym committed:
Update draft-irtf-qirg-principles-03.xml

Co-Authored-By: Wojciech Kozlowski <Wojtek242@users.noreply.github.com>

ngym committed:
Apply suggestions from code review

Co-Authored-By: Wojciech Kozlowski <Wojtek242@users.noreply.github.com>

ngym committed:
Update with Wojtek's comments

ngym committed:
Merge branch 'master' of https://github.com/ngym/draft-irtf-qirg-principles

ngym committed:
separate Pull request for generations

Wojtek242 commented on draft-irtf-qirg-principles-03.xml:
```diff
@@ -795,6 +819,38 @@
+           optimization of quantum networks different from classical ones.
+           <vspace blankLines="1" />
+
+           It is worth noting, that third generation <refer to generations section>
+           quantum repeaters will be able to transmit qubits directly
+           in a "store and forward" manner since they will be able to carry
+           arbitrary states hop by hop. Thus, the networking algorithms of the
+           later generations of quantum networks may end up having more in 
+           common with classical networks.</t>
+           
```
So I've been thinking about this section. Currently it suggests that 3G will likely do store and forward. I think this needs to be relaxed a bit. Or at least made consistent with what I will soon propose with regards to the three generations for the draft. Can we move this to the other pull request too? Rather than merging it now and then changing it. 

Let me know, if you're okay with this and I'll just remove this myself. You'll need to wait for me to merge this request first I guess before you update the other PR.

ngym replied:
No problem. You can remove those sentences by yourself. Please go ahead :)
I'll update the other PR after you merge this request.

Wojtek242 replied:
```suggestion
```

Wojtek242 committed:
Update draft-irtf-qirg-principles-03.xml

Remove generation paragraph so that it can be moved to the PR about repeater generations
